tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp

```diff
@@ -96,16 +97,17 @@ void completion_queue_reserve_back(uint32_t data_size_B) {
         completion_rd_ptr = completion_rd_ptr_and_toggle & 0x7fffffff;
         completion_rd_toggle = completion_rd_ptr_and_toggle >> 31;
     } while (
-        (cq_write_interface.completion_fifo_wr_ptr < completion_rd_ptr) and (cq_write_interface.completion_fifo_wr_ptr + data_size_16B > completion_rd_ptr) or
+        ((cq_write_interface.completion_fifo_wr_ptr < completion_rd_ptr) and (cq_write_interface.completion_fifo_wr_ptr + data_size_16B > completion_rd_ptr)) or
         (completion_rd_toggle != cq_write_interface.completion_fifo_wr_toggle) and (cq_write_interface.completion_fifo_wr_ptr == completion_rd_ptr)
     );
 
     DEBUG_STATUS('N', 'Q', 'R', 'B', 'D');
 }
 
+template <uint32_t host_completion_queue_write_ptr_addr>
 FORCE_INLINE
 void notify_host_of_completion_queue_write_pointer() {
-    constexpr static uint32_t host_completion_queue_write_ptr_addr = get_compile_time_arg_val(0);
+    // constexpr static uint32_t host_completion_queue_write_ptr_addr = get_compile_time_arg_val(0);
```

Comment:
why did you have to make this change?

Reply:
Not a fan of the get_compile call to be in the function. I prefer passing it in from the kernel.